libpromises/dbm_lmdb.c

```diff
@@ -223,11 +227,289 @@ static int LmdbEnvOpen(
     return EBUSY;
 }
 
+/**
+ * @warning Expects @fd_stamp to be locked.
+ */
+static bool RepairedAfterOpen(const char *lmdb_file, int fd_tstamp)
+{
+    time_t repaired_tstamp = -1;
+    ssize_t n_read = read(fd_tstamp, &repaired_tstamp, sizeof(time_t));
+    lseek(fd_tstamp, 0, SEEK_SET);
+    if (n_read == 0)
+    {
+        /* EOF (empty file) => never repaired */
+        Log(LOG_LEVEL_VERBOSE, "DB '%s' never repaired before", lmdb_file);
+    }
+    else if (n_read < sizeof(time_t))
+    {
+        /* error */
+        Log(LOG_LEVEL_ERR, "Failed to read the timestamp of repair of the '%s' DB",
+            lmdb_file);
+    }
+    else
+    {
+        /* read the timestamp => Check if the LMDB file was repaired after
+         * we opened it last time. Or, IOW, if this is a new corruption or
+         * an already-handled one. */
+        DBHandle *handle = GetDBHandleFromFilename(lmdb_file);
+        if (repaired_tstamp > GetDBOpenTimestamp(handle))
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
 static void HandleLDMBCorruption(MDB_env *env, const char *msg)
 {
+    const char *lmdb_file = mdb_env_get_userctx(env);
     Log(LOG_LEVEL_CRIT, "Corruption in the '%s' DB detected! %s",
-        (char *) mdb_env_get_userctx(env), msg);
+        lmdb_file, msg);
+
+#ifdef __MINGW32__
+    /* not much we can do on Windows */
+    Log(LOG_LEVEL_WARNING, "Removing the corrupted DB file '%s'",
```

Comment:
Might be nice to say why there's not much we can do on windows. Is it because fork() is not available/same on Windows?

Reply:
Yes, `fork()` is the biggest problem. Also the file locking on Windows is not so powerful. I'll add a comment.